plugin/Installer/DatabaseChanges.php

```diff
@@ -781,4 +793,27 @@ public function setDetailTemplate()
 		$pDetailView->setTemplate( key( $firstTemplatePath ) );
 		$pDataDetailViewHandler->saveDetailView( $pDetailView );
 	}
+
+
+	/**
+	 *
+	 * @return void
+	 *
+	 */
+
+	public function checkAllPageIdsHaveDetailShortCode()
+	{
+		$pDataDetailViewHandler = $this->_pContainer->get(DataDetailViewHandler::class);
+		$pDetailView            = $pDataDetailViewHandler->getDetailView();
+		$shortCode              = '[oo_estate view="' . $pDetailView->getName() . '"]';
```

Comment:
The shortcode does not always have the form `[oo_estate view="detail"]`. Other variants include:

- `[oo_estate    view="detail"]` (multiple spaces)
- `[oo_estate units="Units" view="detail"]` (with unit lists)

These are not detected, so after the update, the rewrite rule is missing and the detail page inaccessible.

You should change this piece of code to be more general so that any way of writing the shortcode for the detail view is recognized.

Reply:
@jmaas-onoffice Oh, sorry, I forgot about this case, I hardly ever use those shortcodes.

Reply:
We were lucky to have a test instance where this was the case. I also didn't think of this before! :)